(GH-15) Documentation added. (#53)
bbtsoftware/BBT.Maybe: docs/input/docs/getting-started/whymaybe.md

---
Order: 10
Title: Why does BBT.Maybe exist?

> Description: Introduction to BBT.Maybe, what problems are solved and how it is distinctive from other implementations of functional option type.

Reviewer: Something is wrong with the second part of this sentence. Try

```suggestion
Description: Introduction to BBT.Maybe, what problems are solved, and how it's distinctive from other functional option type implementations
```

Author: Adapted.

---

> ## Make optional reference type values explicit

Reviewer: ... and america great again. We should use meaningful titles, instead of this page-wide 2nd level heading I'd rather have multiple headings above short paragraphs, for example

_intro text_
_Features_ (instead of "goals")
_Advantages over similar libraries_

Author: Adapted.

BBT.Maybe is an implementation of the functional option type pattern.

> BBT.Maybe has the following goals:

Reviewer: Aren't the goals automatically in focus of this implementation? See the next paragraph.

Author: Changed according to next paragraph.

* Make nullable references explicit
* Prevent null reference exceptions on reference calls
* Provide an option type with broad usage over all layers of code

> Numerous implementations of functional option type already exist. In focus of this realization are:

Reviewer: You're describing the features of the implementation / library, but not how using _this_ library benefits the user.

Author: I disagree.

* Prevent access to null value
* Implementation of maybe for both optional reference and nullable value types
* Preventing of bloated functionality
* Usage of maybe in communication layer

> Implementation details of maybe are:

Reviewer: _Implementation_ details shouldn't go into the _Why_ section -> Rather _Technical core principles_

Author: Moved implementation details to principles section.
